feat(#84): 답변 댓글ui 구현

Aprasaks opened the pull request:
## 🔧 관련 이슈

- 이 PR은 다음 이슈와 관련 있습니다: #84

## 🔄 변경 사항

<!-- 해당되는 항목만 남기고 삭제하세요 -->

- [x] 기능 추가
- [x] 리팩토링


## ✔️ 변경 사항 상세 설명

Button컴포넌트 타입 추가
commentlist, commentinput, commentitem 
만들어서 answercard에 접목

AnswerCard컴포넌트를 분리작업
CommentList, CommentInput, CommentItem 으로 나눠서 작업
댓글등록시 등록완료 toast 추가

## 📸 스크린샷 (UI 변경 시 필수)

<!-- UI 변경이 있을 경우 스크린샷을 첨부하세요 -->
<!-- 예: ![스크린샷](링크) -->

## 📝 문서화

- [ ] 관련 문서가 업데이트되었습니다. (`README.md`, Notion 링크 등)

## 🔍 리뷰어에게 요청 사항 (선택)

<!-- 리뷰어에게 특별히 확인받고 싶은 부분이 있다면 작성하세요 -->
<!-- 예: 에러 핸들링 방식 괜찮을지 확인 부탁드립니다. -->

## ⚠️ 기타 주의 사항

<!-- 긴급 처리 필요, 병합 순서, 의존성 등 주의 사항이 있다면 작성하세요 -->
<!-- 예: 이 PR은 hotfix이므로 빠른 병합이 필요합니다. -->

Aprasaks committed:
✨ feat: 답변 댓글ui 구현(#84)
변경 사항 : Button컴포넌트에 submit타입추가
AnswerCard컴포넌트를 분리작업
CommentList, CommentInput, CommentItem 으로 나눠서 작업
댓글등록시 등록완료 toast 추가

yunsung-hodoopapa commented on src/components/common/Textarea.tsx:
```diff
@@ -1,37 +1,48 @@
+    const variant = focused ? 'focus' : 'default'
+    const variantMap = {
+      default: 'bg-white text-black border-gray-250',
+      focus: 'border-primary',
```
스타일을 주기 위해 포커스라는 상태가 사용되고 있지만, 
focused, blur는 스타일로도 제어가 가능합니다. 
```suggestion
     className={cn(
          'w-full px-4 py-3 text-sm rounded-lg transition-colors resize-none border focus:outline-none',
          'bg-white text-black border-gray-250',
          'focus:border-primary',
          className
        )}
```

Aprasaks replied:
확실히 불필요한 상태값이 사라지네요

yunsung-hodoopapa commented on src/components/qna/AnswerCard.tsx:
```diff
@@ -1,136 +1,258 @@
-import { MessageCircle } from 'lucide-react'
+// import { MessageCircle } from 'lucide-react'
```
주석 아래 코드가 깔려있는데, 이 주석이 develop으로 병합되면 안됩니다.
기억해두세용 develop은 모두를 위한 작업 공간입니다. 불필요한 코드는 꼭 제거하고 올려주세요

yunsung-hodoopapa commented on src/components/qna/CommentInput.tsx:
```diff
@@ -0,0 +1,86 @@
+const NOTICE =
+  '개인정보를 공유 및 요청하거나, 명예 훼손, 무단 광고, 불법 정보 유포시 모니터링 후 삭제될 수 있습니다.'
+const DEFAULT_MAX_LENGTH = 300
+
```
상수 선언

컴포넌트 인터페이스 

컴포넌트 순으로 위치시키는 것이 훨씬 좋습니다 :-)

yunsung-hodoopapa commented on src/components/qna/CommentList.tsx:
```diff
@@ -0,0 +1,36 @@
+  className?: string
+}
+
+function CommentList({ comments, className = '' }: CommentListProps) {
```
```suggestion
function CommentList({ comments = [], className = '' }: CommentListProps) {
```

yunsung-hodoopapa commented on src/components/qna/CommentList.tsx:
```diff
@@ -0,0 +1,36 @@
+}
+
+interface CommentListProps {
+  comments: Comment[]
```
comments?: Comment[]

Aprasaks committed:
✨ feat: 답변등록ui review.v(#84)

변경 사항 : comments에 빈배열추가로 undefined 사전방지
상수선언 그리고 interface 그리고 컴포넌트 순
테일윈드로 스타일제어->불필요한 상태관리 삭제

Aprasaks committed:
✨ feat: 답변 채택하기기능 구현(#84)
변경 사항 : 질문상세데이터mock 데이터 구현
현재 로그인된 유저의 id와 nickname, student를 기준으로
채택하기가 뜨고 안뜨고를 구현
채택하기를 누르면 모든 답변에서 채택하기버튼이사라지고
채택한 답변만이 변하게끔 수정

devjaesung reviewed (approved):
수고하셨습니다! 강혁님!
muy bien!

jngmnj commented on src/components/common/Textarea.tsx:
```diff
@@ -1,37 +1,37 @@
-}: TextareaProps) {
-  return (
-    <div className="w-full">
+const Textarea = forwardRef<HTMLTextAreaElement, TextareaProps>(
```
배워갑니다!! 

## 왜 forwardRef를 사용하는가?
기본적으로 React 컴포넌트는 자신의 내부 DOM을 외부에서 직접 접근하지 못하게 되어 있습니다. 그런데 아래와 같은 경우에는 외부에서 컴포넌트 내부의 DOM 요소에 직접 접근해야 할 필요가 생깁니다:

예시 상황:
- 외부에서 해당 <textarea>에 자동 포커스를 주고 싶을 때
- 외부에서 스크롤 조정, 높이 측정 등을 하고 싶을 때
- formRef.current.focus() 같이 DOM API를 직접 써야 할 때

이럴 때, forwardRef를 사용하면 부모 컴포넌트에서 자식 컴포넌트 내부의 DOM에 접근 가능하게 됩니다.
